### ISSUE TITLE
Ambigous Method Invocation for Boxed / Non-Boxed Types

### ISSUE THREAD

**Issue by bytefish** [JSqlServerBulkInsert/JSqlServerBulkInsert#19]
I have a problem with the current API design, which also made me rewrite some functions in PgBulkInsert. We support both writing ``mapInteger`` with an ``Integer`` and a primitive ``int``. This is done by using a custom Lambda "ToInt" function. But this leads to the ugly side-effect, that the JVM cannot resolve the correct method:

![image](https://user-images.githubusercontent.com/239373/107768878-66304380-6d37-11eb-8e69-976adade2279.png)

Basically I would like to rename the methods to:

* ``mapBooleanPrimitive``
* ``mapFloatPrimitive``
* ...

So this would not happen anymore. And one can easily choose to use the boxed / unboxed type.

**Comment by bytefish** [comment]
The reason to change it is also related to Issue #17. All columns have to be mapped, so there should be a simple way to assign default values to the column. If we are renaming to "mapIntegerPrimitive" for example, I can easily write something like this in a column:

![image](https://user-images.githubusercontent.com/239373/107769422-43525f00-6d38-11eb-89b8-2de602bdba81.png)
